src/main/kotlin/com/chattriggers/ctjs/minecraft/objects/gui/Gui.kt

```diff
@@ -241,33 +241,130 @@ abstract class Gui : GuiScreen() {
     /**
      * Add a base Minecraft button to the gui
      *
-     * @param buttonId   id for the button
-     * @param x          X position of the button
-     * @param y          Y position of the button
-     * @param buttonText the label of the button
+     * @param button the button to add
+     * @return the Gui for method chaining
      */
-    fun addButton(buttonId: Int, x: Int, y: Int, buttonText: String) {
-        buttons.add(GuiButton(buttonId, x, y, buttonText))
-    }
+    fun addButton(button: GuiButton) = apply { buttons.add(button) }
 
     /**
      * Add a base Minecraft button to the gui
      *
-     * @param buttonId   id for the button
-     * @param x          X position of the button
-     * @param y          Y position of the button
-     * @param width      the width of the button
-     * @param height     the height of the button
+     * @param buttonId id for the button
+     * @param x X position of the button
+     * @param y Y position of the button
+     * @param width the width of the button
+     * @param height the height of the button
      * @param buttonText the label of the button
+     * @return the Gui for method chaining
      */
-    fun addButton(buttonId: Int, x: Int, y: Int, width: Int = 200, height: Int = 20, buttonText: String) {
+    @JvmOverloads
+    fun addButton(buttonId: Int, x: Int, y: Int, width: Int = 200, height: Int = 20, buttonText: String) = apply {
         buttons.add(GuiButton(buttonId, x, y, width, height, buttonText))
     }
 
-    fun setButtonVisibility(buttonId: Int, visible: Boolean) {
-        buttons.firstOrNull {
-            it.id == buttonId
-        }?.visible = visible
+    /**
+     * Removes a button from the gui with the given id
+     *
+     * @param buttonId the id of the button to remove
+     * @return the Gui for method chaining
+     */
+    fun removeButton(buttonId: Int) = apply {
+        buttons.removeIf { it.id == buttonId }
+    }
+
+    fun clearButtons() = apply { buttons.clear() }
+
+    fun getButton(buttonId: Int): GuiButton? = buttons.firstOrNull { it.id == buttonId }
+
+    fun getButtonVisibility(buttonId: Int): Boolean = getButton(buttonId)?.visible ?: false
+
+    /**
+     * Sets the visibility of a button
+     *
+     * @param buttonId the id of the button to change
+     * @param visible the new visibility of the button
+     * @return the GuiButton for method chaining
+     */
+    fun setButtonVisibility(buttonId: Int, visible: Boolean) = getButton(buttonId)?.apply {
```

Comment:
This should return the `Gui` rather than the button itself. Applies for all of the setters below.